tests/_resources/versioned_app/app.py

```diff
@@ -34,7 +34,8 @@ async def lifespan(app: FastAPI):
 versioned_app_with_custom_api_version_var.add_unversioned_routers(webhooks_router)
 
 client = TestClient(versioned_app, raise_server_exceptions=False, headers=BASIC_HEADERS)
-client_without_headers = TestClient(versioned_app)
+with TestClient(versioned_app) as client_without_headers:
+    pass
```

Comment:
Seems like this change is unnecessary

```suggestion
client_without_headers = TestClient(versioned_app)
```

Reply:
I don't like it either, but the startup event will not be triggered for TestClient if the object does not act as a context manager.

```bash
2 failed
         - tests/test_applications.py:93 test__get_openapi
         - tests/test_applications.py:107 test__get_docs__all_versions
```
when run the tests with:

```bash
client_without_headers = TestClient(versioned_app)
```